### PR TITLE
Add description and parameters to expression suggestions

### PR DESCRIPTION
**Pull request by bartektartanus**

### CODE REVIEW

**Review by arkadius — APPROVED** [review]

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala [hunk 1]
```diff
@@ -51,20 +51,20 @@ class SpelExpressionSuggester(expressionConfig: ExpressionDefinition[_], typeDef
...
+          .map(_.map(list => list.map(e => ExpressionSuggestion(e.label, td, fromClass = false, None, Nil)))).getOrElse(successfulNil)
       }.getOrElse(successfulNil)
     }
 
     def filterClassMethods(classDefinition: ClazzDefinition, name: String, staticContext: Boolean, fromClass: Boolean = false): List[ExpressionSuggestion] = {
       val methods = filterMapByName(if (staticContext) classDefinition.staticMethods else classDefinition.methods, name)
 
       methods.values.flatten
-        .map(m => ExpressionSuggestion(m.name, m.signatures.head.result, fromClass = fromClass))
+        .map(m => ExpressionSuggestion(m.name, m.signatures.head.result, fromClass = fromClass, m.description, m.signatures.head.noVarArgs.map(p => Parameter(p.name, p.refClazz))))
```
Comment (arkadius) [inline interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala:67]: Shouldn't be `m.signatures.toList.maxBy(_.parametersToList.length)` as it is in `UIProcessObjectsFactory.toUIMethod`? Also let's add `TODO: present all overloaded methods on FE`

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala [hunk 2]
```diff
@@ -215,4 +215,6 @@ private case class NuSpelNodeParent(node: NuSpelNode, nodeIndex: Int)
 // TODO: fromClass is used to calculate suggestion score - to show fields first, then class methods. Maybe we should
 //  return score from BE?
 @JsonCodec(encodeOnly = true)
-case class ExpressionSuggestion(methodName: String, refClazz: TypingResult, fromClass: Boolean)
+case class ExpressionSuggestion(methodName: String, refClazz: TypingResult, fromClass: Boolean, description: Option[String], parameters: List[Parameter])
+@JsonCodec(encodeOnly = true)
+case class Parameter(name: String, refClazz: TypingResult)
```
[the remark this answers is not in the document]
Comment (bartektartanus) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala:220]: Yes, but this class is in `restmodel` and is not visible in `interpreter` package.

[the remark this answers is not in the document]
Comment (arkadius) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala:220]: Good point, let's leave it as it is

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala (hunk 1 above)
Comment (bartektartanus) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala:67]: 1. Ok, changed that.
2. Sure, done that.
3. Not sure about that. The only common code is: `signatures.toList.maxBy(_.parametersToList.length)` and `signature.noVarArgs ::: signature.varArg.toList` and we intend to change that to show all methods.
4. Done

Comment (arkadius) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala]: ok

**Pull request merged**